platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMapOptions.java

```diff
@@ -217,7 +218,7 @@ public static MapboxMapOptions createFromAttributes(@NonNull Context context, @N
       Drawable compassDrawable = typedArray.getDrawable(
         R.styleable.mapbox_MapView_mapbox_uiCompassDrawable);
       if (compassDrawable == null) {
-        compassDrawable = ContextCompat.getDrawable(context, R.drawable.mapbox_compass_icon);
+        compassDrawable = ResourcesCompat.getDrawable(context.getResources(), R.drawable.mapbox_compass_icon, null);
```

Comment:
You're right. I messed `getDrawable` APIs. [`Resources#getDrawable()`](https://developer.android.com/reference/android/content/res/Resources.html#getDrawable%28int%29) was the one deprecated. Actually the way to go is `ContextCompat` as pointed out [here](https://plus.google.com/+BenjaminWeiss/posts/M1dYFaobyBM).
Funny thing is that `ResourcesCompat` isn't deprecated 😂 
What a mess! First world problems of having multiple classes doing the same...